index.d.ts

```diff
@@ -4,16 +4,20 @@ export type AllowedCardNetworkType = 'AMEX' | 'DISCOVER' | 'JCB' | 'MASTERCARD'
 
 export type AllowedCardAuthMethodsType = 'PAN_ONLY' | 'CRYPTOGRAM_3DS'
 
+export type tokenizationSpecificationType = 'PAYMENT_GATEWAY' | 'DIRECT'
+
 export interface RequestDataType {
   cardPaymentMethod: {
     tokenizationSpecification: {
-      type: string
-      gateway: string
-      gatewayMerchantId: string
+      type: tokenizationSpecificationType
+      gateway?: string
+      gatewayMerchantId?: string
       stripe?: {
         publishableKey: string
         version: string
       }
+      /** only for DIRECT */
+      publicKey?: string
     }
```

Comment:
May be it's better not to put comment here, but better generate JSDoc for this interface? Then it'll be obvious with which type which parameters should go(also gateway and merchant id should be only with PAYMENT_GATEWAY which you didn't state here in comment)